ui/packages/tidb-dashboard-lib/src/apps/Deadlock/pages/Detail.tsx

```diff
@@ -0,0 +1,81 @@
+import React, { useContext, useState } from 'react'
+import { DeadlockModel } from '@lib/client'
+import { useLocation } from 'react-router-dom'
+import { useEffectOnce } from 'react-use'
+import { useTranslation } from 'react-i18next'
+
+import { CardTable, HighlightSQL } from '@lib/components'
+import { CacheContext } from '@lib/utils/useCache'
+import DeadlockChainGraph from '../components/DeadlockChainGraph'
+import { DeadlockContext } from '../context'
+
+function Detail() {
+  const ctx = useContext(DeadlockContext)
+  const { t } = useTranslation()
+  const cache = useContext(CacheContext)
+  const id = new URLSearchParams(useLocation().search).get('id')
+  let [isLoading, setIsLoading] = useState(true)
+  let [items, setItems] = useState([] as DeadlockModel[])
+  useEffectOnce(() => {
+    setIsLoading(true)
+    if (cache?.get(`deadlock-${id}`) !== undefined) {
+      setItems(cache.get(`deadlock-${id}`))
+      setIsLoading(false)
+    } else {
+      ctx!.ds.deadlockListGet().then(({ data }) => {
+        data.forEach((it) => {
+          let items = cache?.get(`deadlock-${it.id}`) || []
+          items.push(it)
+          cache?.set(`deadlock-${it.id}`, items)
+        })
+        setItems(data.filter((it) => it.id?.toString() === id))
+        setIsLoading(false)
+      })
+    }
+  })
+  return (
+    <>
+      <DeadlockChainGraph
+        deadlockChain={items}
+        onHover={(id: string) => {
+          console.log(id)
+        }}
+      />
+      <CardTable
+        loading={isLoading}
+        columns={[
+          {
+            name: t('deadlock.fields.try_lock_trx_id'),
+            key: 'try_lock_trx_id',
+            minWidth: 100,
+            onRender: (it) => it.try_lock_trx_id
+          },
+          {
+            name: t('deadlock.fields.current_sql'),
+            key: 'current_sql',
+            minWidth: 350,
+            onRender: (it) => <HighlightSQL sql={it.current_sql} compact />
+          },
+          {
+            name: t('deadlock.fields.key'),
+            key: 'key',
+            minWidth: 300,
+            onRender: (it) => it.key
+          },
+          {
+            name: t('deadlock.fields.trx_holding_lock'),
+            key: 'trx_holding_lock',
+            minWidth: 150,
+            onRender: (it) => it.trx_holding_lock
+          }
+        ]}
```

Comment:
I think it is better to define the `columns` variable in front, to make the component props concise.